### DISCUSSION TITLE
server/client not working on ubuntu16.04

### DISCUSSION THREAD

**Discussion by truewq** [minhanghuang/CyberRT#9] [Q&A] [answered]
Hi @minhanghuang 
I built the cyberRT on ubuntu16.04. I did the server/client test, but it's not working. Do you have any suggestion?
![image](https://user-images.githubusercontent.com/7168949/206410320-fce0e45b-8ab5-4f99-a861-ded605165175.png)

The ./cyber/examples/cyber_example_server does not receive request from ./cyber/examples/cyber_example_cilent

I checked the service and node, seems client_node status is wrong. 

`test@test-desktop:/xxxxxx/cyberRT/CyberRT/build$ cyber_service list
WARNING: Logging before InitGoogleLogging() is written to STDERR
I20221208 17:32:56.115265 30192 class_loader_utility.h:79] []registerclass:Circle,Base,
I20221208 17:32:56.115352 30192 class_loader_utility.h:79] []registerclass:Rect,Base,
I20221208 17:32:56.115362 30192 class_loader_utility.h:79] []registerclass:Triangle,Base,
I20221208 17:32:56.115383 30192 class_loader_utility.h:79] []registerclass:Star,Base,
I20221208 17:32:56.115393 30192 class_loader_utility.h:79] []registerclass:Apple,Base,
I20221208 17:32:56.115401 30192 class_loader_utility.h:79] []registerclass:Pear,Base,
I20221208 17:32:56.115408 30192 class_loader_utility.h:79] []registerclass:Banana,Base,
I20221208 17:32:56.115428 30192 class_loader_utility.h:79] []registerclass:Peach,Base,
The number of services is:  1
example_server
test@test-desktop:/xxxxxx/cyberRT/CyberRT/build$ cyber_node list
WARNING: Logging before InitGoogleLogging() is written to STDERR
I20221208 17:33:04.397598 30216 class_loader_utility.h:79] []registerclass:Circle,Base,
I20221208 17:33:04.397670 30216 class_loader_utility.h:79] []registerclass:Rect,Base,
I20221208 17:33:04.397679 30216 class_loader_utility.h:79] []registerclass:Triangle,Base,
I20221208 17:33:04.397686 30216 class_loader_utility.h:79] []registerclass:Star,Base,
I20221208 17:33:04.397696 30216 class_loader_utility.h:79] []registerclass:Apple,Base,
I20221208 17:33:04.397701 30216 class_loader_utility.h:79] []registerclass:Pear,Base,
I20221208 17:33:04.397720 30216 class_loader_utility.h:79] []registerclass:Banana,Base,
I20221208 17:33:04.397740 30216 class_loader_utility.h:79] []registerclass:Peach,Base,
Number of active nodes: 2
cilent_node
server_node
test@test-desktop:/xxxxx/cyberRT/CyberRT/build$ cyber_node info client node
WARNING: Logging before InitGoogleLogging() is written to STDERR
I20221208 17:33:13.596354 30271 class_loader_utility.h:79] []registerclass:Circle,Base,
I20221208 17:33:13.596416 30271 class_loader_utility.h:79] []registerclass:Rect,Base,
I20221208 17:33:13.596442 30271 class_loader_utility.h:79] []registerclass:Triangle,Base,
I20221208 17:33:13.596452 30271 class_loader_utility.h:79] []registerclass:Star,Base,
I20221208 17:33:13.596463 30271 class_loader_utility.h:79] []registerclass:Apple,Base,
I20221208 17:33:13.596472 30271 class_loader_utility.h:79] []registerclass:Pear,Base,
I20221208 17:33:13.596494 30271 class_loader_utility.h:79] []registerclass:Banana,Base,
I20221208 17:33:13.596504 30271 class_loader_utility.h:79] []registerclass:Peach,Base,
E20221208 17:33:15.602447 30271 py_cyber.h:454] [cyber_py]no node named: client
RoleAttributes ParseFromString failed. size is  0
E20221208 17:33:17.602814 30271 py_cyber.h:454] [cyber_py]no node named: node
RoleAttributes ParseFromString failed. size is  0
test@test-desktop:/xxxxxx/cyberRT/CyberRT/build$ cyber_node info server_node
WARNING: Logging before InitGoogleLogging() is written to STDERR
I20221208 17:33:28.656095 30307 class_loader_utility.h:79] []registerclass:Circle,Base,
I20221208 17:33:28.656170 30307 class_loader_utility.h:79] []registerclass:Rect,Base,
I20221208 17:33:28.656180 30307 class_loader_utility.h:79] []registerclass:Triangle,Base,
I20221208 17:33:28.656188 30307 class_loader_utility.h:79] []registerclass:Star,Base,
I20221208 17:33:28.656196 30307 class_loader_utility.h:79] []registerclass:Apple,Base,
I20221208 17:33:28.656203 30307 class_loader_utility.h:79] []registerclass:Pear,Base,
I20221208 17:33:28.656210 30307 class_loader_utility.h:79] []registerclass:Banana,Base,
I20221208 17:33:28.656217 30307 class_loader_utility.h:79] []registerclass:Peach,Base,
Node:    	server_node
ProcessId: 	29622
Hostname:	test-desktop
[Reading Channels]:

[Writing Channels]:

`

Hi 大佬黄
    我测试server/client时，发现client与server的通信异常。用cyber_xx脚本检查，发现client的节点信息查不到。请教一下定位问题的建议思路。

**Comment by minhanghuang** [comment, marked as answer]
> Hi @minhanghuang I built the cyberRT on ubuntu16.04. I did the server/client test, but it's not working. Do you have any suggestion? ![image](https://user-images.githubusercontent.com/7168949/206410320-fce0e45b-8ab5-4f99-a861-ded605165175.png)
> 
> The ./cyber/examples/cyber_example_server does not receive request from ./cyber/examples/cyber_example_cilent
> 
> I checked the service and node, seems client_node status is wrong.
> 
> `test@test-desktop:/xxxxxx/cyberRT/CyberRT/build$ cyber_service list WARNING: Logging before InitGoogleLogging() is written to STDERR I20221208 17:32:56.115265 30192 class_loader_utility.h:79] []registerclass:Circle,Base, I20221208 17:32:56.115352 30192 class_loader_utility.h:79] []registerclass:Rect,Base, I20221208 17:32:56.115362 30192 class_loader_utility.h:79] []registerclass:Triangle,Base, I20221208 17:32:56.115383 30192 class_loader_utility.h:79] []registerclass:Star,Base, I20221208 17:32:56.115393 30192 class_loader_utility.h:79] []registerclass:Apple,Base, I20221208 17:32:56.115401 30192 class_loader_utility.h:79] []registerclass:Pear,Base, I20221208 17:32:56.115408 30192 class_loader_utility.h:79] []registerclass:Banana,Base, I20221208 17:32:56.115428 30192 class_loader_utility.h:79] []registerclass:Peach,Base, The number of services is: 1 example_server test@test-desktop:/xxxxxx/cyberRT/CyberRT/build$ cyber_node list WARNING: Logging before InitGoogleLogging() is written to STDERR I20221208 17:33:04.397598 30216 class_loader_utility.h:79] []registerclass:Circle,Base, I20221208 17:33:04.397670 30216 class_loader_utility.h:79] []registerclass:Rect,Base, I20221208 17:33:04.397679 30216 class_loader_utility.h:79] []registerclass:Triangle,Base, I20221208 17:33:04.397686 30216 class_loader_utility.h:79] []registerclass:Star,Base, I20221208 17:33:04.397696 30216 class_loader_utility.h:79] []registerclass:Apple,Base, I20221208 17:33:04.397701 30216 class_loader_utility.h:79] []registerclass:Pear,Base, I20221208 17:33:04.397720 30216 class_loader_utility.h:79] []registerclass:Banana,Base, I20221208 17:33:04.397740 30216 class_loader_utility.h:79] []registerclass:Peach,Base, Number of active nodes: 2 cilent_node server_node test@test-desktop:/xxxxx/cyberRT/CyberRT/build$ cyber_node info client node WARNING: Logging before InitGoogleLogging() is written to STDERR I20221208 17:33:13.596354 30271 class_loader_utility.h:79] []registerclass:Circle,Base, I20221208 17:33:13.596416 30271 class_loader_utility.h:79] []registerclass:Rect,Base, I20221208 17:33:13.596442 30271 class_loader_utility.h:79] []registerclass:Triangle,Base, I20221208 17:33:13.596452 30271 class_loader_utility.h:79] []registerclass:Star,Base, I20221208 17:33:13.596463 30271 class_loader_utility.h:79] []registerclass:Apple,Base, I20221208 17:33:13.596472 30271 class_loader_utility.h:79] []registerclass:Pear,Base, I20221208 17:33:13.596494 30271 class_loader_utility.h:79] []registerclass:Banana,Base, I20221208 17:33:13.596504 30271 class_loader_utility.h:79] []registerclass:Peach,Base, E20221208 17:33:15.602447 30271 py_cyber.h:454] [cyber_py]no node named: client RoleAttributes ParseFromString failed. size is 0 E20221208 17:33:17.602814 30271 py_cyber.h:454] [cyber_py]no node named: node RoleAttributes ParseFromString failed. size is 0 test@test-desktop:/xxxxxx/cyberRT/CyberRT/build$ cyber_node info server_node WARNING: Logging before InitGoogleLogging() is written to STDERR I20221208 17:33:28.656095 30307 class_loader_utility.h:79] []registerclass:Circle,Base, I20221208 17:33:28.656170 30307 class_loader_utility.h:79] []registerclass:Rect,Base, I20221208 17:33:28.656180 30307 class_loader_utility.h:79] []registerclass:Triangle,Base, I20221208 17:33:28.656188 30307 class_loader_utility.h:79] []registerclass:Star,Base, I20221208 17:33:28.656196 30307 class_loader_utility.h:79] []registerclass:Apple,Base, I20221208 17:33:28.656203 30307 class_loader_utility.h:79] []registerclass:Pear,Base, I20221208 17:33:28.656210 30307 class_loader_utility.h:79] []registerclass:Banana,Base, I20221208 17:33:28.656217 30307 class_loader_utility.h:79] []registerclass:Peach,Base, Node: server_node ProcessId: 29622 Hostname: test-desktop [Reading Channels]:
> 
> [Writing Channels]:
> 
> `
> 
> Hi 大佬黄 我测试server/client时，发现client与server的通信异常。用cyber_xx脚本检查，发现client的节点信息查不到。请教一下定位问题的建议思路。

Hi @truewq   会不会是`test@test-desktop:/xxxxx/cyberRT/CyberRT/build$ cyber_node info client node`  中间遗漏`_`的造成的? 我不确定你当时出入的命令  可以试试`cyber_node info client_node`查看节点信息

**Reply by truewq** [reply]
HI @minhanghuang 

确实是拼写错误，cyber_node可以正常显示：
/cyberRT/CyberRT/build$ cyber_node info cilent_node
WARNING: Logging before InitGoogleLogging() is written to STDERR
I20221208 19:58:48.014986 32064 class_loader_utility.h:79] []registerclass:Circle,Base,
I20221208 19:58:48.015075 32064 class_loader_utility.h:79] []registerclass:Rect,Base,
I20221208 19:58:48.015101 32064 class_loader_utility.h:79] []registerclass:Triangle,Base,
I20221208 19:58:48.015110 32064 class_loader_utility.h:79] []registerclass:Star,Base,
I20221208 19:58:48.015118 32064 class_loader_utility.h:79] []registerclass:Apple,Base,
I20221208 19:58:48.015125 32064 class_loader_utility.h:79] []registerclass:Pear,Base,
I20221208 19:58:48.015146 32064 class_loader_utility.h:79] []registerclass:Banana,Base,
I20221208 19:58:48.015158 32064 class_loader_utility.h:79] []registerclass:Peach,Base,
Node:    	cilent_node
ProcessId: 	31905
Hostname:	test-desktop
[Reading Channels]:

[Writing Channels]:


但是，cilent与server之间的通信还是异常的……

**Reply by minhanghuang** [reply]
> HI @minhanghuang
> 
> 确实是拼写错误，cyber_node可以正常显示： /cyberRT/CyberRT/build$ cyber_node info cilent_node WARNING: Logging before InitGoogleLogging() is written to STDERR I20221208 19:58:48.014986 32064 class_loader_utility.h:79] []registerclass:Circle,Base, I20221208 19:58:48.015075 32064 class_loader_utility.h:79] []registerclass:Rect,Base, I20221208 19:58:48.015101 32064 class_loader_utility.h:79] []registerclass:Triangle,Base, I20221208 19:58:48.015110 32064 class_loader_utility.h:79] []registerclass:Star,Base, I20221208 19:58:48.015118 32064 class_loader_utility.h:79] []registerclass:Apple,Base, I20221208 19:58:48.015125 32064 class_loader_utility.h:79] []registerclass:Pear,Base, I20221208 19:58:48.015146 32064 class_loader_utility.h:79] []registerclass:Banana,Base, I20221208 19:58:48.015158 32064 class_loader_utility.h:79] []registerclass:Peach,Base, Node: cilent_node ProcessId: 31905 Hostname: test-desktop [Reading Channels]:
> 
> [Writing Channels]:
> 
> 但是，cilent与server之间的通信还是异常的……
![image](https://user-images.githubusercontent.com/37317982/206632918-3dc24e17-6aa0-42c0-ae46-3f403a13c608.png)

需要配一下日志的路径, 你可以把example里的`AINFO`改成`std::cout`
![image](https://user-images.githubusercontent.com/37317982/206634326-50254b4a-c42c-4a3d-8b42-6dae98945825.png)
